lib/nautic_net/serial/server.ex

```diff
@@ -0,0 +1,59 @@
+defmodule NauticNet.Serial.Server do
+  use GenServer
+  require Logger
+
+  def start_link(false), do: :ignore
+
+  def start_link(config) do
+    GenServer.start_link(__MODULE__, config, name: __MODULE__)
+  end
+
+  @impl GenServer
+  def init(config) do
+    Logger.debug("Serial config: #{inspect(config)}")
+    # Open Port
+    {driver, driver_config} = get_callback_module_config(config[:driver] || raise(":driver config is required"))
+
+    # Start listening
+    :ok = driver.init(driver_config)
+
+    # Start Handlers
+    handlers =
+      get_callback_module_configs(config[:handlers] || raise(":handlers config is required"))
+      |> Enum.map(fn {handler, handler_config} ->
+        {handler, handler.init(handler_config)}
+      end)
+
+    {:ok, %{driver: driver, handlers: handlers, port_name: "ttyUSB1"}}
+  end
+
+  defp get_callback_module_config({module, config}), do: {module, config}
+  defp get_callback_module_config(module), do: {module, []}
+
+  defp get_callback_module_configs(list) do
+    Enum.map(list, &get_callback_module_config/1)
+  end
+
+  @impl GenServer
+
+  def handle_info({:circuits_uart, port_name, sentence}, %{port_name: port_name} = state) do
+    case NMEA.to_data(:nmea_0183, sentence) do
+      :invalid ->
+        nil
+
+      params ->
+        for p <- params do
+          data = %NMEA.Data{
+            values: p,
```

Comment:
This field should be renamed `:value` if it is only going to store a single params struct.

Reply:
I went back and forth on this... because a single params struct (the argument for `:value`) has multiple values (the argument for `:values`) so I went back and forth a lot and decided to stick with values. With that in mind do you have a strong preference... totally open to discuss.

Reply:
Maybe just `:params`?